cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java

```diff
@@ -243,4 +295,27 @@ public URI toURI() {
     // Not used.
     return null;
   }
+
+  private static String encodeFormatSpec(FormatSpecification formatSpecification) {
+    String asJson = GSON.toJson(formatSpecification);
+    try {
+      return URLEncoder.encode(asJson, "UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      // this should never happen
+      throw Throwables.propagate(e);
+    }
+  }
+
+  private static FormatSpecification decodeFormatSpec(String encodedFormatSpecification) {
+    if (encodedFormatSpecification == null) {
+      return null;
+    }
+    try {
+      String decodedFormatSpecification = URLDecoder.decode(encodedFormatSpecification, "UTF-8");
+      return GSON.fromJson(decodedFormatSpecification, FormatSpecification.class);
+    } catch (UnsupportedEncodingException e) {
+      // shouldn't ever happen
+      return null;
```

Comment:
Although it won't happen, it'd be good to have unify way to handle exception (you propagate above but return null in here).